### PR TITLE
fix(filter): support filtering empty arrays

### PR DESCRIPTION
**Pull request by skjindal93**

### CODE REVIEW

File: entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java [hunk 1]
```diff
@@ -31,8 +33,8 @@ public FilterTypeExpression convert(
       final LiteralConstant constant,
       final RequestContext requestContext)
       throws ConversionException {
-    final String id = columnIdentifier.getColumnName();
-    final String subDocPath = getSubDocPathById(entityAttributeMapping, id, requestContext);
+    final String columnId = columnIdentifier.getColumnName();
+    final String subDocPath = getSubDocPath(columnId, requestContext);
 
     final IdentifierExpression identifierExpression = IdentifierExpression.of(subDocPath);
```
Comment (suresh-prakash) [inline entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java:39]: Can be simplified after injecting org.hypertrace.entity.query.service.converter.identifier.IdentifierExpressionConverter. 
```suggestion
    final IdentifierExpression identifierExpression = identifierExpressionConverter.convert(columnIdentifier);
```

File: entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java [hunk 2]
```diff
@@ -45,16 +47,37 @@ public FilterTypeExpression convert(
...
+  /**
+   * An array is stored as 'attributes.array_field.valueList.values.%d.value.&lt;type&gt;'.
+   *
+   * <p>Example: products = ['shoes', 'socks'] is stored as
+   * 'attributes.names.valueList.values.0.value.string: shoes', and
+   * 'attributes.names.valueList.values.1.value.string: socks'.
+   *
+   * <p>An empty array is stored as 'attributes.array_field.valueList: {}'
+   */
+  private String getSubDocPath(String columnId, RequestContext requestContext)
```
Comment (suresh-prakash) [inline entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java:74]: Once the above comment is addressed, we could get rid of these 2 methods as the same business logic is being done in org.hypertrace.entity.query.service.converter.identifier.IdentifierExpressionConverter

File: entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java (hunk 1 above)
Comment (skjindal93) [reply entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java:39]: Yes. That's a nice suggestion!

File: entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java [hunk 3]
```diff
@@ -10,18 +8,16 @@
...
+import org.hypertrace.entity.query.service.converter.identifier.IdentifierExpressionConverter;
 import org.hypertrace.entity.query.service.v1.ColumnIdentifier;
 import org.hypertrace.entity.query.service.v1.LiteralConstant;
 import org.hypertrace.entity.query.service.v1.Operator;
 
 @AllArgsConstructor(onConstructor_ = {@Inject})
 public class NullFilteringExpressionConverter extends FilteringExpressionConverterBase {
-  private final EntityAttributeMapping entityAttributeMapping;
-  private final IdentifierConverterFactory identifierConverterFactory;
+  private final IdentifierExpressionConverter identifierExpressionConverter;
```
Comment (suresh-prakash) [inline entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java:20]: Is this binding working? I see that IdentifierExpressionConverter is bound to `Converter<ColumnIdentifier, IdentifierExpression>` only @ IdentifierModule

**Review by suresh-prakash — APPROVED** [review]

File: entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java (hunk 3 above)
Comment (skjindal93) [reply entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java:20]: fair point. let me test this end to end

Comment (skjindal93) [reply entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java:20]: tested this e2e. seems to be working fine

**Pull request merged**